packages/jest-environment-jsdom/src/index.js

```diff
@@ -44,13 +44,32 @@ class JSDOMEnvironment {
       };
     }
 
+    // Report uncaught errors.
     this.errorEventListener = event => {
-      if (event.error) {
+      if (userErrorListenerCount === 0 && event.error) {
```

Comment:
Instead of manual bookkeeping, can we just check the total number of listeners here?

e.g. this works in chrome:

```js
window.getEventListeners(window).error.length // 1 on github.com
```

Reply:
Looks like `getEventListeners ` is not available for jsdom@9 at least

Reply:
I'm fine with upgrading to 11 and dropping node 4 now.

Reply:
Seems to be a devtools API, so never mind!

https://developers.google.com/web/tools/chrome-devtools/console/command-line-reference#geteventlistenersobject

Reply:
#4769